### PR TITLE
Unify timestamp

### PR DESCRIPTION
**Pull request by sin6pi7**
- Add Timestamp type
- Use Timestamp in backend
- Use Timestamp in state and encoding
- Final touches

### CODE REVIEW

File: packages/backend/src/peripherals/database/RollupStateRepository.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export class RollupStateRepository implements IRollupStateStorage {
 
 function parametersToJson(parameters: RollupParameters) {
   return {
-    timestamp: parameters.timestamp.toString(),
+    timestamp: BigInt(Number(parameters.timestamp)),
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/RollupStateRepository.ts:134]: There are no bigints in json. This has to be a number then

File: packages/encoding/test/decodeOnChainData.test.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import REAL_DECODED from './data/onchain-decoded.json'
 import REAL_DATA from './data/onchain-example.json'
 
 describe('decodeOnChainData', () => {
-  it('decodes real onchain data', () => {
+  it('decodes real onchain data', async () => {
```
Comment (sz-piotr) [inline packages/encoding/test/decodeOnChainData.test.ts:8]: why?

File: packages/state/src/RollupState.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export class RollupState {
           throw new Error('Invalid update count')
         }
         const funding =
-          update.fundingTimestamp !== 0n
+          Number(update.fundingTimestamp) !== 0
```
Comment (sz-piotr) [inline packages/state/src/RollupState.ts:60]: Maybe `Timestamp(0)` would be cleaner here?

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/backend/src/peripherals/database/RollupStateRepository.ts (hunk 1 above)
Comment (sin6pi7) [reply packages/backend/src/peripherals/database/RollupStateRepository.ts:134]: Returned type is not really a json - `timestamp` is of type `bigint` and `funding` of `jsonb`. I think naming here is confusing, so I changed `json` to `row`, which reflects reality.

File: packages/encoding/test/decodeOnChainData.test.ts (hunk 2 above)
Comment (sin6pi7) [reply packages/encoding/test/decodeOnChainData.test.ts:8]: That's on me - forgot to clean this up - removed now.

File: packages/state/src/RollupState.ts (hunk 3 above)
Comment (sin6pi7) [reply packages/state/src/RollupState.ts:60]: You're right - done.

**Review by sz-piotr — APPROVED** [review]

**Pull request merged**